
# Alignment with minimap2

For alignment, the pipeline uses minimap2 on each FASTQ file. The reads are then sorted by genomic coordinates, and an integrity check is performed on the resulting BAM file.

## Aligning and Sorting

##### Align and sort reads

```text
sentieon minimap2 -Y -L --eqx --secondary=no -ax map-ont reference.fasta reads.fastq |
  samtools sort -o sorted.bam -
```

Arguments:

- *-Y*: enable soft clipping.
- *-L*: use long cigars for tag `CG`.
- *-\-eqx*: use X/= in cigars instead of M.
- *-\-secondary=no*: ignore secondary alignments.

## Integrity Check

To confirm the integrity of the alignment BAM file, in-house Python code checks for the presence of the 28-byte empty block representing the EOF marker in BAM format.

## Implementation with Sentieon

Sentieon implementation replicates the original [minimap2](https://github.com/lh3/minimap2) code. The pipeline is using Sentieon version 202308.01, corresponding to minimap2 version 2.26.

## Source Code

All the relevant code can be accessed in the GitHub repository:

  - [sentieon_minimap2_sort.sh](https://github.com/smaht-dac/sentieon-pipelines/blob/main/dockerfiles/sentieon/sentieon_minimap2_sort.sh) [minimap2]

---

<!--> This section relies on the html links generated by GitHub Pages 
and will not render correctly in Markdown <-->
[_Home_](/pipelines-docs/) - [_Overview_](0_Overview.html) - [_Read Groups_](2_Read_Groups.html) - [_Methylation and Tags_](3_Methylation_and_Tags.html)
